### PR TITLE
refactor(localdebug): update localdebug plugin and UX(ext&cli) to use localSettings

### PR DESCRIPTION
**Pull request by dooriya**
## Changes:
If `TEAMSFX_MULTI_ENV` feature flag is enabled:
- LocalDebug plugin will use `localSettings` to read/write configurations for local debug.
- The `local.env` file will not be used in `post-localdebug` to persist the environment variables passed to UX layer. Instead, we maintain these local envs in-memory, and expose an user task to invoke by the toolkit extension.
- Extension and CLI will use the `localSetings.json` file  as the source-of-truth to read configuration for local debug.

Otherwise:
- The local debug behavior keeps the same as original, which still uses `env.default.json`, `local.env`, `default.userdata` to run local debug.

**NOTE**: To make this PR more friendly to be reviewed, I copy the [original `localDebug` & `postLocalDebug`](https://github.com/OfficeDev/TeamsFx/blob/dev/packages/fx-core/src/plugins/resource/localdebug/index.ts#L193) lifecycle API implementation to `legacyPlugin.ts`, so you can skip reviewing this file @kuojianlu @swatDong.

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/localdebug/index.ts [hunk 1]
```diff
@@ -51,13 +51,25 @@ import { getCodespaceName, getCodespaceUrl } from "./util/codespace";
...
+  LocalSettingsBackendKeys,
+  LocalSettingsBotKeys,
+  LocalSettingsFrontendKeys,
+  LocalSettingsTeamsAppKeys,
+} from "../../../common/localSettingsConstants";
 @Service(ResourcePlugins.LocalDebugPlugin)
 export class LocalDebugPlugin implements Plugin {
   name = "fx-resource-local-debug";
   displayName = "LocalDebug";
+  localEnvs: { [key: string]: string } = {};
```
[the remark this answers is not in the document]
Comment (dooriya) [reply packages/fx-core/src/plugins/resource/localdebug/index.ts:67]: Previously, I already use `getLocalDebugEnvs` to generate the envs. And according to the behaviors in current Teams toolkit, I found that these local envs only generated in the `postLocalDebug` lifecycle, and might be frequent invoked in UX (e.g. the extension task provider call it to resolve the debug tasks). So I move to a property and avoid already generate the envs when calling `getLocalDebugEnvs`.

[the remark this answers is not in the document]
Comment (swatDong) [reply packages/fx-core/src/plugins/resource/localdebug/index.ts:67]: There's user scenario to run "Start Xxx" task only without debug... so generating envs in `postLocalDebug` does not provide the latest variables.

BTW, I only see `getLocalDebugEnvs` in `postLocalDebug` which seems redundant?

[the remark this answers is not in the document]
Comment (dooriya) [reply packages/fx-core/src/plugins/resource/localdebug/index.ts:67]: Yes, I only generated the envs once in `postLocalDebug` and simply return the `localEnvs` property when calling the `getLocalDebugEnvs` user task.
But as you mentioned that there's scenario that user can  run "Start Xxx" task without debug, I should change back to always return the latest envs by calling `getLocalDebugEnvs`.

[the remark this answers is not in the document]
Comment (dooriya) [reply packages/fx-core/src/plugins/resource/localdebug/index.ts]: Resolved in this commit: [fix: always return the updated local envs in getLocalDebugEnvs](https://github.com/OfficeDev/TeamsFx/pull/1902/commits/660b495d559b6497ea80a9b96e6279bd2bceebaf)

**Review by kuojianlu — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Review by 1yefuwang1 — APPROVED** [review]

**Pull request merged**